assets/app/view/game/round/stock.rb

```diff
@@ -357,12 +357,9 @@ def render_buy_input_interval(company)
 
         def render_bid_input(company)
           return [] if @step.max_bid(@current_entity, company) < @step.min_bid(company)
-          return [] if @current_actions.include?('bidding_tokens') &&
-            @step.highest_player_bid?(@current_entity, company)
-          return [] if @current_actions.include?('bidding_tokens') && @step.find_bid(@current_entity, company).nil? &&
-            @step.bidding_tokens(@current_entity).zero?
+          return [] if @current_actions.include?('bidding_tokens') && !@step.can_bid?(@current_entity, company)
```

Comment:
Why do we need the bidding_tokens action? Can't we use bid? Isn't the player card the only place in the UI that needs to know about bidding_tokens?

Reply:
Refactor line 359 to use `can_bid?` and 1822 bidding is common with other game bidding within the UI hooray.

Reply:
The can_bid? method isnt in the normal Auctioner module which uses the 'bid' action. Any game that uses bid in the stockround will then throw an error when trying to access can_bid?.  bidding_tokens is also used in the player.rb to see how many tokens you have left to bid. Dont know how to show them otherwise?

Reply:
if we cant use @step.respond_to?(:can_bid?) then if we gonna use bid, we need to have a dummy method in Auctioner with can_bid? not to break for every other game.

Reply:
What I meant by refactor is move line 359 to be a method called `can_bid?` within Auctioner. Keep the logic and game differentiation to the engine code as much as possible. Working on the UI can often include a bunch of refactoring to more generalize the code to include your use cases.

See my other comment on the player card.

Reply:
I can remove bidding token on the player and print a log message if this is a better solution.

Reply:
Sure you can count your bids, as well. Just wanted the user to see in one place. Committed cash, Available cash and how many tokens he/she have left.

Reply:
Ok i see i have made lots of comments without seeing your new ones :)

Reply:
Definitely keep the bidding tokens on the player card. Its important info that they shouldn't have to calculate for each player.